### PR TITLE
Work around NumPy issue with `dtype.kind` of unsigned ints

### PR DESCRIPTION
**Pull request by johnkerl**
Finding:

```
>>> np.issubdtype(np.asarray([1,2,3],dtype=np.int32).dtype, np.integer)
True

>>> np.issubdtype(np.asarray([1,2,3],dtype=np.uint32).dtype, np.integer)
True

>>> np.issubdtype(np.asarray([1,2,3],dtype=np.int32).dtype.kind, np.integer)
True

>>> np.issubdtype(np.asarray([1,2,3],dtype=np.uint32).dtype.kind, np.integer)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/numpy/core/numerictypes.py", line 416, in issubdtype
    arg1 = dtype(arg1).type
TypeError: data type 'u' not understood
```

So we should just do the first two, not the second two.

Found by @bkmartinjr in a TileDB-SOMA context: https://gist.github.com/johnkerl/3beaf565041a4acdbd534cba0e53bdae

I'll turn that gist into a TileDB-SOMA-Py unit-test case.

### CODE REVIEW

**Review by nguyenv — COMMENTED** [review]
Thank you for catching and correcting this. I will approve once I see all green. Unfortunately CI is running really slowly today.

**Comment by johnkerl** [comment]
[sc-35950]

**Comment by nguyenv** [comment]
Unit test errors have been corrected in https://github.com/TileDB-Inc/TileDB-Py/pull/1858. This branch needs to be rebased against `main` once the fix has been merged.

**Comment by johnkerl** [comment]
At the moment, this is rebased atop #1858

**Comment by johnkerl** [comment]
[sc-35946]

**Comment by johnkerl** [comment]
@nguyenv CI is green 🙏

**Review by nguyenv — APPROVED** [review]

**Comment by johnkerl** [comment]
@nguyenv I lack merge permissions on this repo -- can you merge please?

**Comment by johnkerl** [comment]
Thanks @nguyenv !!

**Pull request merged**
